starter-core/src/main/java/io/micronaut/starter/feature/github/workflows/docker/templates/gradleGraalCEWorkflow.rocker.raw

```diff
@@ -0,0 +1,42 @@
+@import io.micronaut.starter.application.Project
+
+@args (
+Project project,
+String graalVersion
+)
+
+name: GraalVM CE CI
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@@v2
+      - uses: actions/cache@@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.m2/repository
+          key: ${{ runner.os }}-gradle-test-${{ hashFiles('**/*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-test-
+      - name: Setup GraalVM CE
+        uses: DeLaGuardo/setup-graalvm@@8bbfe44ef9c6f5c07e5af036a1bffd561c037d18
+        with:
+            graalvm-version: @graalVersion
+      - name: Install Native Image
+        run: gu install native-image
+      - name: Docker login
+        uses: docker/login-action@@v1
+        with:
+          registry: ${{ secrets.DOCKER_REGISTRY_URL }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push docker image
+        env:
+          DOCKER_ORGANIZATION: ${{ secrets.DOCKER_ORGANIZATION }}
+          DOCKER_REGISTRY_URL: ${{ secrets.DOCKER_REGISTRY_URL }}
+          TESTCONTAINERS_RYUK_DISABLED: true
+        export DOCKER_IMAGE=`echo "${DOCKER_REGISTRY_URL}/${DOCKER_ORGANIZATION}/@project.getName()" | sed -e 's#//#/#' -e 's#^/##'`
```

Comment:
By default we want to support ideally all possible docker registries. For this purpose two secrets (`DOCKER_REGISTRY_URL` and `DOCKER_ORGANIZATION`) are created from which the final DOCKER_IMAGE name is composed, plust the DOCKER_REGISTRY_URL is used for login into docker registry.

One could argue that simple concatenation like `$DOCKER_REGISTRY_URL/$DOCKER_ORGANIZATION/$project.name` inside pom.xml or build.gradle would be enough. Unfortunately the opposite is the truth because such chaining would cause creation of invalid docker image name:
- dockerhub - no need to specify uri, so the final image is `/pgressa/foo`
- amazon ecr - in case of flat docker registry hierarchy , the DOCKER_ORGANIZATION is not set, so the final image is `<aws-account-number>.dkr.ecr.<region>.amazonaws.com//foo:0.1`

B/c of the above the final image name is easier to be done in bash than in gradle/pom.

Reply:
Does this actually work? looks like invalid YAML

Reply:
true.. missed this one, the final test suite is still running

Reply:
for review purpose I'll generate all 4 combinations and reference it from description so you can check